mesh/mesh_readers.cpp

```diff
@@ -383,47 +383,49 @@ void Mesh::CreateVTKMesh(const Vector &points, const Array<int> &cell_data,
    int order = -1;
    bool legacy_elem = false, lagrange_elem = false;
 
-   int j = 0;
-   for (int i = 0; i < NumOfElements; i++)
    {
-      int ct = cell_types[i];
-      Geometry::Type geom = VTKGeometry::GetMFEMGeometry(ct);
-      elements[i] = NewElement(geom);
-      if (cell_attributes.Size() > 0)
-      {
-         elements[i]->SetAttribute(cell_attributes[i]);
-      }
-      // VTK ordering of vertices is the same as MFEM ordering of vertices
-      // for all element types *except* prisms, which require a permutation
-      if (geom == Geometry::PRISM && ct != VTKGeometry::LAGRANGE_PRISM)
+      int j = 0;
+      for (int i = 0; i < NumOfElements; i++)
       {
-         int prism_vertices[6];
-         for (int k=0; k<6; ++k)
+         int ct = cell_types[i];
+         Geometry::Type geom = VTKGeometry::GetMFEMGeometry(ct);
+         elements[i] = NewElement(geom);
+         if (cell_attributes.Size() > 0)
          {
-            prism_vertices[k] = cell_data[j+VTKGeometry::PrismMap[k]];
+            elements[i]->SetAttribute(cell_attributes[i]);
+         }
+         // VTK ordering of vertices is the same as MFEM ordering of vertices
+         // for all element types *except* prisms, which require a permutation
+         if (geom == Geometry::PRISM && ct != VTKGeometry::LAGRANGE_PRISM)
+         {
+            int prism_vertices[6];
+            for (int k=0; k<6; ++k)
+            {
+               prism_vertices[k] = cell_data[j+VTKGeometry::PrismMap[k]];
+            }
+            elements[i]->SetVertices(prism_vertices);
+         }
+         else
+         {
+            elements[i]->SetVertices(&cell_data[j]);
          }
-         elements[i]->SetVertices(prism_vertices);
-      }
-      else
-      {
-         elements[i]->SetVertices(&cell_data[j]);
-      }
 
-      int elem_dim = Geometry::Dimension[geom];
-      int elem_order = VTKGeometry::GetOrder(ct, cell_offsets[i] - j);
+         int elem_dim = Geometry::Dimension[geom];
+         int elem_order = VTKGeometry::GetOrder(ct, cell_offsets[i] - j);
```

Comment:
Rather than adding a new scope, we could just do something like
```c++
int j = (i > 0) ? cell_offsets[i-1] : 0;
```
at the beginning of the loop body.

Reply:
Done